internal/engine/upper.go

```diff
@@ -486,32 +495,66 @@ func handleStartProfilingAction(state *store.EngineState) {
 	state.IsProfiling = true
 }
 
-func handleFSEvent(
-	ctx context.Context,
-	state *store.EngineState,
-	event fswatch.TargetFilesChangedAction) {
+func handleFileWatchCreateEvent(_ context.Context, state *store.EngineState, action fswatch.FileWatchCreateAction) {
+	name := types.NamespacedName{Namespace: action.FileWatch.GetNamespace(), Name: action.FileWatch.GetName()}
+	state.FileWatches[name] = action.FileWatch
+}
 
-	if event.TargetID.Type == model.TargetTypeConfigs {
-		for _, f := range event.Files {
-			state.PendingConfigFileChanges[f] = event.Time
+func handleFileWatchUpdateEvent(_ context.Context, state *store.EngineState, action fswatch.FileWatchUpdateAction) {
```

Comment:
fwiw, i ended up moving the Handle functions into the subpackage. which made them easier to use in tests...